src/App.tsx

```diff
@@ -14,11 +14,118 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  goods: string[],
+  isVisible: boolean,
+  start: boolean,
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    goods: [...goodsFromServer],
+    isVisible: false,
+    start: true,
+  };
+
+  reverseGoods = (goods: string[]) => {
+    return goods.reverse();
+  };
+
+  render() {
+    const { goods, isVisible, start } = this.state;
+
+    return (
+      <div className="app">
+        <h1 className="app__title">Goods</h1>
+        {start && (
+          <button
+            type="button"
+            onClick={() => {
+              this.setState({ isVisible: true });
+              this.setState({ start: false });
+              this.setState({ goods: [...goodsFromServer] });
+            }}
+            className="app__start-button"
+          >
+            Start
+          </button>
+        )}
+
+        {isVisible && (
+          <div className="app__container">
+            <div className="app__finish-button">
+              <button
+                type="button"
+                onClick={() => {
+                  this.setState({ isVisible: false });
+                  this.setState({ start: true });
+                }}
+                className="button__finish-button"
+              >
+                Finish
+              </button>
+            </div>
+
+            <div className="app__content">
+              <ul className="app__list">
+                {goods.map(good => (
+                  <li key={good} className="app__item">
+                    {good}
+                  </li>
+                ))}
+              </ul>
+            </div>
+
+            <div className="app__buttons">
+              <button
+                type="button"
+                onClick={() => {
+                  this.setState({
+                    goods: goods.sort((a, b) => a.localeCompare(b)),
```

Comment:
sort is a mutable method, make a copy before using sort

Reply:
i disagree. i have had made array copy in state. and mutate it only in one session. when we push "reset" ore "finish" button, we are change mutated array for new copy.